sbt-dumpschema-plugin/src/main/scala/org/virtuslab/akkasaferserializer/SchemaWriter.scala

```diff
@@ -0,0 +1,43 @@
+package org.virtuslab.akkasaferserializer
+
+import better.files.File
+import io.bullet.borer.Json
+import org.virtuslab.akkasaferserializer.model.{Codecs, TypeDefinition}
+
+import scala.collection.mutable
+
+class SchemaWriter(outputDirectory: File) extends Codecs {
+
+  def this(dumpSchemaOptions: DumpSchemaOptions) = {
+    this(File(dumpSchemaOptions.outputDir))
+  }
+
+  lazy val lastDump: Map[String, TypeDefinition] = {
+    outputDirectory
+      .createDirectoryIfNotExists(createParents = true)
+      .list(_.extension.contains(".json"))
+      .flatMap { file =>
+        Json.decode(file.byteArray).to[TypeDefinition].valueTry.toOption
+      }
+      .map(x => (x.name, x))
+      .toMap
+  }
+
+  private val dumped: mutable.Set[String] = mutable.Set()
```

Comment:
Statefulness 😱

Reply:
... can it be avoided somehow here?....

Reply:
Well...
I need to cache disc operations somehow but it's impossible to achieve stateless cache with one source file as a function entrance point.
I consider caching disc operations as more important than statelessness, but I am open to ideas on how to mitigate it.